src/orchestrator-process/src/lib.rs

```diff
@@ -271,7 +290,6 @@ impl NamespacedOrchestrator for NamespacedProcessOrchestrator {
                 ))));
             }
         }
-        supervisors.insert(id.into(), handles);
```

Comment:
It's taken care of here: https://github.com/MaterializeInc/materialize/pull/14148/files#diff-818d26f34fdb3ab6b0892e482b0a1231780a2321f7981fc96912e022bc1d6a6dR161

Instead of deleting the old handles and inserting the new one, it modifies the existing one in place. Saves a lookup, but mostly I just found it easier to manage when I was rewriting this to handle re-ensuring an existing service.